tapir/configuration/models.py

```diff
@@ -0,0 +1,69 @@
+from enum import Enum
+
+from django.db import models
+
+
+class ParameterTag(models.Model):
+    tag = models.CharField(max_length=64, primary_key=True)
+
+    def __str__(self):
+        return self.tag
+
+
+class ParameterDatatype(Enum):
+    INTEGER = "integer"
+    DECIMAL = "decimal"
+    BOOLEAN = "boolean"
+    STRING = "string"
+
+
+class Parameter(models.Model):
+    key = models.CharField(max_length=256, primary_key=True, editable=False)
+    description = models.CharField(max_length=512)
+    tags = models.ManyToManyField(ParameterTag)
+    datatype = models.CharField(
+        choices=map(lambda d: (d, d), ParameterDatatype), max_length=8, editable=False
+    )
+    value = models.CharField(max_length=4096, null=True)
+
+    def get_value(self):
+        if self.datatype == ParameterDatatype.INTEGER.value:
+            return int(self.value)
+        elif self.datatype == ParameterDatatype.DECIMAL.value:
+            return float(self.value)
+        elif self.datatype == ParameterDatatype.BOOLEAN.value:
+            return bool(self.value)
+        elif self.datatype == ParameterDatatype.STRING.value:
+            return self.value
+        else:
+            Exception("""Unknown parameter type: {}""".format(self.datatype))
+
+
+def get_config(key: Enum):
+    param = Parameter.objects.get(pk=key.value)
+    return param.get_value()
+
+
+def parameter_definition(
+    key: Enum,
+    description: str,
+    tags: list,
+    datatype: ParameterDatatype,
+    initial_value: str,
+):
+    converted_tags = list(map(lambda t: ParameterTag(t), tags))
+    for tag in converted_tags:
+        tag.save()
+
+    param = Parameter(
+        key=key.value,
+        description=description,
+        tags=converted_tags,
+        datatype=datatype,
+        value=initial_value,
+    )
+    param.save(
+        update_fields=["description", "tags", "datatype"]
+    )  # don't update value from the definition
```

Comment:
Here I would also do something in the form of "check if the parameter exists, if yes update it, otherwise create it" to be more explicit.

Reply:
OK I can do that, but isn't save() doing exactly that?

https://docs.djangoproject.com/en/3.1/ref/models/instances/#how-django-knows-to-update-vs-insert